components/menu/sub_menu.tsx

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode, useState, MouseEvent} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import MuiMenu, {MenuProps as MuiMenuProps} from '@mui/material/Menu';
+import MuiMenuList from '@mui/material/MenuList';
+import MuiMenuItem from '@mui/material/MenuItem';
+import {PopoverOrigin} from '@mui/material/Popover';
+import {styled as muiStyled} from '@mui/material/styles';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import {getIsMobileView} from 'selectors/views/browser';
+
+import {openModal, closeModal} from 'actions/views/modals';
+
+import CompassDesignProvider from 'components/compass_design_provider';
+import GenericModal from 'components/generic_modal';
+
+interface Props {
+
+    // Anchor button props
+    triggerId?: string;
+    triggerElement?: ReactNode;
+    triggerAriaLabel?: string;
+
+    // Menu props
+    menuId: string;
+    menuAriaLabel?: string;
+    openAt?: 'right' | 'left';
+
+    children: ReactNode;
+
+}
+
+export function SubMenu(props: Props) {
+    const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+    const isSubMenuOpen = Boolean(anchorElement);
+
+    const isMobileView = useSelector(getIsMobileView);
+
+    const theme = useSelector(getTheme);
+
+    const dispatch = useDispatch();
+
+    const handleSubMenuOpen = (event: MouseEvent<HTMLLIElement>) => {
+        event.preventDefault();
+        setAnchorElement(event.currentTarget);
+    };
+
+    const handleSubMenuClose = (event: MouseEvent<HTMLLIElement>) => {
+        event.preventDefault();
+        setAnchorElement(null);
+    };
+
+    const hasSubmenuItems = Boolean(props.children);
+    if (!hasSubmenuItems) {
+        return null;
+    }
+
+    if (isMobileView) {
+        function MenuModalComponent() {
```

Comment:
Same comment as with the `Menu` here. I think we could move all the mobile stuff into its own function

Reply:
Oops missed this one. Yes moved to separate component.